pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -288,31 +290,25 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 
 // Check if cluster name is changed in hub kubeconfig secret by comparing with deployment args. Force update the deployment
 // if the secret is changed.
-func (n *klusterletController) forceRollOutWorkAgent(ctx context.Context, config klusterletConfig, forceRollOut *bool) {
-	if *forceRollOut {
-		return
-	}
-
+func (n *klusterletController) forceRollOutWorkAgent(ctx context.Context, config klusterletConfig) bool {
 	workDeploymentName := fmt.Sprintf("%s-work-agent", config.KlusterletName)
 	workDeployment, err := n.kubeClient.AppsV1().Deployments(config.KlusterletNamespace).Get(ctx, workDeploymentName, metav1.GetOptions{})
 	if err != nil {
-		return
+		return false
```

Comment:
shouldn't this be true? If we didn't find it, it should be rolled out, right?

Reply:
I think we will create it in applyDeployment  if it is not found in the following. I will add a comment here.